WEB_SERVICE/WEB/_MEDIUM/HTTP_METHOD_MANIPULATION/recommendation.md

```diff
@@ -0,0 +1,22 @@
+To mitigate the risks associated with **HTTP Method Manipulation** in GraphQL, follow these security practices:
+
+1. **Enforce POST-only Mutations**: Ensure that mutations can only be executed via the POST method. Reject any mutation requests made using GET to prevent sensitive information from being passed in URLs.
+  
+2. **Disable GET for Mutations**: Update server configurations to explicitly disallow mutations over GET requests. This ensures that no data-altering operations can be performed via a URL.
+
+3. **Use Secure Proxies**: If a proxy is used to route requests, ensure it does not log sensitive URLs, or implement logging sanitization to remove sensitive information from the logs.
+
+4. **Monitor and Test Regularly**: Continuously test GraphQL endpoints for method manipulation vulnerabilities. Ensure that mutation requests can only be executed via the appropriate HTTP method.
+
+```python
+# Enforce POST-only mutations in a Flask-based GraphQL app
+
+from flask import request, jsonify
+from flask_graphql import GraphQLView
+
+@app.route('/graphql', methods=['POST'])
+def graphql():
+    if request.method != 'POST':
+        return jsonify({"error": "Only POST requests allowed for mutations"}), 405
+
+    return GraphQLView.as_view('graphql')()
```

Comment:
```
=== "Python"
  ```python
  # Enforce POST-only mutations in a Flask-based GraphQL app

  from flask import request, jsonify
  from flask_graphql import GraphQLView

  @app.route('/graphql', methods=['POST'])
  def graphql():
      if request.method != 'POST':
          return jsonify({"error": "Only POST requests allowed for mutations"}), 405

      return GraphQLView.as_view('graphql')()
```
      
So that it looks like this in the UI:  
![image](https://github.com/user-attachments/assets/6450e12c-f805-4054-973e-c39e04159e31)

Same for the code block in `description.md`